cogs/hideout/boost_roles.py

```diff
@@ -0,0 +1,265 @@
+            except ValueError:
+                pass
+
+        if icon and emoji:
+            return await interaction.response.send_message("You can not supply both `icon` and `emoji`.", ephemeral=True)
+
+        if emoji is not None:
+            conv = await self.get_emoji(emoji)
+            if conv is None:
+                return await interaction.response.send_message("Could not parse that emoji.", ephemeral=True)
+
+            icon_ = conv
+
+        elif icon is not None:
+            if icon.size > 256 * 10**3:  # 256 kB
+                return await interaction.response.send_message("The icon size needs to be less than 256kB.", ephemeral=True)
+
+            icon_ = await icon.read()
+
+        role = interaction.guild.get_role(db["role_id"])
+        assert role
+
+        if icon_ is not None:
+            role = await role.edit(
+                name=name or MISSING, colour=colour_ if colour_ != discord.Colour.default() else MISSING, display_icon=icon_
+            )
+
+        else:
+            role = await role.edit(
+                name=name or MISSING,
+                colour=colour_ if colour_ != discord.Colour.default() else MISSING,
+            )
+
+        assert role
+
+        query = """
+            UPDATE booster_roles
+                SET role_name = $1,
+                    role_colour = $2,
+                    role_icon = $3
+                WHERE role_id = $4
+            RETURNING *
+        """
+
+        db = await interaction.client.pool.fetchrow(
+            query, role.name, str(role.colour), role.icon and await role.icon.read(), role.id
+        )
+
+        assert db
+
+        embed: Optional[discord.Embed] = MISSING
+        role_icon: Optional[discord.File] = MISSING
+        if role.icon is not None:
+            role_icon = discord.File(filename=f"icon.png", fp=BytesIO(db["role_icon"]))
+            embed = discord.Embed().set_thumbnail(url="attachment://icon.png")
+
+        text = f"Successfully edited {role.mention}" f"\nName: {role}" f"\nColor: {db['role_colour'].upper()}"
+        await interaction.response.send_message(text, file=role_icon or MISSING, embed=embed or MISSING)
+
+    @role.command()
+    async def delete(self, interaction: Interaction):
+        """Deletes your boost role."""
+        db = await interaction.client.pool.fetchrow("SELECT * FROM booster_roles WHERE user_id = $1", interaction.user.id)
+
+        if db is None:
+            return await interaction.response.send_message("You don't have a boost role.", ephemeral=True)
+
+        assert interaction.guild
+
+        role = interaction.guild.get_role(db["role_id"])
+        assert role
+
+        await role.delete(reason=f"Nitro Boost Expired: {interaction.user}")
+
+        await interaction.client.pool.execute("DELETE FROM booster_roles WHERE user_id = $1", interaction.user.id)
+        await interaction.response.send_message("Successfully deleted your boost role.")
```

Comment:
add an `on_guild_role_delete` handler for if someone manually deletes one of these too. A simple blind DELETE is fine, without checking stuff or anything.